ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManager.java

```diff
@@ -212,6 +212,12 @@ public static ReactInstanceManagerBuilder builder() {
     return new ReactInstanceManagerBuilder();
   }
 
+  /** Creates a builder that is capable of creating an instance of {@link ReactInstanceManager}.
+   * and specifies weather to load with hermes or jsc */
+  public static ReactInstanceManagerBuilder builder(Boolean hermesEnabled) {
```

Comment:
Isn't better if we use an enum instead of a boolean flag?

We can have an:
```java
enum JSInterpreter {
  OLD_LOGIC,
  JSC,
  HERMES
}
```
And switch over it. It will improve readability, it avoids possible bugs due to the use of a null value which has unclear semantic and it is ready if we would ever add a new interpreter later on.

Reply:
Hi @cipolleschi , the Boolean is passed via the host app's BuildConfig. Hence keeping it null makes much more sense as no argument would automatically imply to use the old logic.

Reply:
@KunalFarmah98 I understand your point, though using an enum here is a bit more future proof.

Can I ask you to update the `ReactInstanceManagerBuilder` to:
1. Remove the constructor which accepts a boolean in input
1. Add a `setJsEngine(JSEngine)` method that accepts the enum
2. Add a `setJsEngineAsHermes(boolean)` parameter which will if true `setJsEngine(HERMES)` else `setJsEngine(JSC)`?

Reply:
Hi @cortinico.
I have made the requested changes. I will update the PR description in some time. Meanwhile you can review the new changes.

Reply:
PR description is updated. For some reason, android CI tests have started failing but I do not see any errors in Android Studio. Please let me know if I have missed something here.

Reply:
Thanks for doing it @KunalFarmah98 👍 Left a minor comment, and we should be good to import/merge it